File: tests/libpyre/journal/info.cc
// -*- coding: utf-8 -*-
//
// michael a.g. aïvázis
// california institute of technology
// (c) 1998-2012 all rights reserved
//


// for the build system
#include <portinfo>

// packages
#include <cassert>
// access to the journal header file
#include <pyre/journal.h>

// main program
int main() {

    // instantiate a info channel
    pyre::journal::info_t info("pyre.journal.test");
    // check that it is inactive, by default
    assert(info.isActive() == false);

    // activate it
    info.activate();
    // and check
    assert(info.isActive() == true);

    // now, instantiate again using the same channel name
    pyre::journal::info_t again("pyre.journal.test");
    // check that it is active
    assert(again.isActive() == true);

    // all done
    return 0;
}

// end of file

File: tests/journal/info.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis
# california institute of technology
# (c) 1998-2012 all rights reserved
#


"""
Verify that info channels by the same name share a common state
"""


def test():
    # access the package
    import journal
    # build a info channel
    info = journal.info("journal.test1")
    # verify that it is off by default
    assert info.active == False
    # enable it
    info.active = True

    # access the same channel through another object
    clone = journal.info("journal.test1")
    # verify that it is now on
    assert clone.active == True

    # build a info channel with a different name
    another = journal.info("journal.test2")
    # verify that it is off by default, to make sure that there is no crosstalk between channels
    assert another.active == False

    # all done
    return


# main
if __name__ == "__main__":
    test()


# end of file 
